shared/js/blobview.js

```diff
@@ -420,33 +333,36 @@ var BlobView = (function() {
     },
 
     // Read bytes up to and including a null terminator, but never
-    // more than size bytes.  And return as a Latin1 string
-    readNullTerminatedLatin1Text: function(size) {
-      var s = '';
-      for (var i = 0; i < size; i++) {
-        var charcode = this.view.getUint8(this.index + i);
-        if (charcode === 0) {
-          i++;
-          break;
-        }
-        s += String.fromCharCode(charcode);
+    // more than size bytes.  And return as an ASCII string
+    readNullTerminatedASCIIText: function(size) {
+      var bytes = new Uint8Array(this.buffer, this.viewOffset + this.index,
+                                 size);
```

Comment:
What happens if size extends beyond the end of the buffer? Do both implementations fail in that case?